docs/deployment.md

```diff
@@ -23,6 +23,25 @@ For example, pre-pulling images to reduce the core deployment utility runtime.
   6. Run deployment utility
   7. Post deploy checks
 
+### Features and Profiles
+
+To allow deployments with different sets of functionality enabled, the deployment utility supports features and profiles.
+
+- A feature is an abstract representation of "the deployed system can now do X", usually implemented by enabling a Foreman/Pulp/Hammer plugin (or a collection of these). 
+- A profile is a set of features that are enabled by default and can not be disabled. This is to allow common deployment types like "vanilla foreman", "katello", "satellite" and similar.
+
+Additionally to the functionality offered by plugins, we define the following "base" features:
+- `foreman` to deploy the main Rails app and make the deployment a "Server"
+- `foreman-proxy` to deploy the Foreman Proxy code
+- `hammer` to deploy the base CLI
+
+These base features control which plugins are enabled when a feature is requested.
```

Comment:
"when a feature is requested" -> Did you mean to say "profile" here?

If not, what'd be a valid "profile" at this point?

Reply:
This PR only defines one: katello.
Others might be foreman, satellite, proxy, content-proxy, capsule